src/wallet/walletdb.cpp

```diff
@@ -1303,3 +1327,15 @@ bool CWalletDB::WriteHDChain(const CHDChain& chain)
     nWalletDBUpdated++;
     return Write(std::string("hdchain"), chain);
 }
+
+uint256 SproutKeyChecksum(const libzcash::SproutPaymentAddress& addr, const libzcash::SproutSpendingKey& key)
+{
+    uint256 hashSprout = addr.GetHash();
+    return Hash(hashSprout.begin(), hashSprout.end(), key.begin(), key.end());
+}
+
+uint256 SaplingKeyChecksum(const libzcash::SaplingIncomingViewingKey& ivk,
+                           const libzcash::SaplingExtendedSpendingKey& key)
+{
+    return Hash(ivk.begin(), ivk.end(), BEGIN(key), END(key));
```

Comment:
`BEGIN()` and `END()` appear to be incompatible with the `libzcash::SaplingExtendedSpendingKey` for purposes of hashing. The `libzcash::SaplingExtendedSpendingKey` struct uses an 8 bit int for the depth, causing the memory of the entire struct to be padded. Subtracting the begin ptr from the end ptr gives 172 bytes. Adding the data members manually by eye gives 169 bytes. Showing that it was padded by 3 bytes because of the 8 bit data member.

I really thought it would be nice here to not copy secret key data into a datastream for hashing. If that is kept a priority, then it seems like the next step would be to add a `libzcash::SaplingExtendedSpendingKey::GetHash()` function that hashes together the data internally (I did this on my local code and that solves the problem).

Suggestions?